subworkflows/local/utils_nfcore_phaseimpute_pipeline/main.nf

```diff
@@ -287,9 +305,44 @@ def validateInputParameters() {
     assert params.step, "A step must be provided"
 
     // Check that at least one tool is provided
-    if (params.step.split(',').contains("impute") || params.step.split(',').contains("panelprep")) {
+    if (params.step.split(',').contains("impute")) {
         assert params.tools, "No tools provided"
     }
+
+    // Check that input is provided for all steps, except panelprep
+    if (params.step.split(',').contains("all") || params.step.split(',').contains("impute") || params.step.split(',').contains("simulate") || params.step.split(',').contains("validate")) {
```

Comment:
To simplify the usage of params.tools and params.step we could modify it at the start of the initialisation:
`params.tools = params.tools.split(',')` ?
We should also rename params.step to params.steps.